test_image_without_pillow cannot find pypdf

Running pytest in a test version of pypdf when pypdf is not installed in pyenv.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
# Linux-6.8.0-41-generic-x86_64-with-glibc2.34

$ python -c "import pypdf;print(pypdf._debug_versions)"
# pypdf==4.3.1, crypt_provider=('pycryptodome', '3.19.0'), PIL=9.5.0
```

## Code

This is a minimal, complete example that shows the issue. Ensure that pypdf is not installed in pyenv then:

```python
# pytest tests/test_filters.py::test_image_without_pillow
```

## Traceback

This is the complete traceback I see:

```
>       assert result.returncode == 0
E       assert 1 == 0
E        +  where 1 = CompletedProcess(args=['/home/kaos/.pyenv/versions/3.8.2/bin/python', PosixPath('/tmp/pytest-of-kaos/pytest-20/test_im...w0/script.py", line 3, in <module>\n    from pypdf import PdfReader\nModuleNotFoundError: No module named \'pypdf\'\n').returncode
```

The test generates a script and runs that script, which changes sys.path to:

['/tmp/pytest-of-kaos/pytest-24/test_image_without_pillow0',
  '/home/kaos/.pyenv/versions/3.8.2/lib/python38.zip',
  '/home/kaos/.pyenv/versions/3.8.2/lib/python3.8',
  '/home/kaos/.pyenv/versions/3.8.2/lib/python3.8/lib-dynload',
  '/home/kaos/.pyenv/versions/3.8.2/lib/python3.8/site-packages']

One of two things happens:
1) pypdf is found in pyenv so the script runs, but is not testing the version in the current directory.
2) The only copy of pypdf is the current directory which is not on sys.path, so the script fails.

Easy fix: Add "." to the start of env PYTHONPATH so the test runs the correct version of pypdf from the local directory, not from pyenv.

--- a/tests/test_filters.py
+++ b/tests/test_filters.py
@@ -248,6 +248,7 @@ def test_issue_399():
 
 @pytest.mark.enable_socket()
 def test_image_without_pillow(tmp_path):
+    import os
     name = "tika-914102.pdf"
     pdf_path = Path(__file__).parent / "pdf_cache" / name
     pdf_path_str = str(pdf_path.resolve()).replace("\\", "/")
@@ -273,9 +274,15 @@ for page in reader.pages:
     ), exc.value.args[0]
 """
     )
+    env = os.environ.copy()
+    try:
+        env["PYTHONPATH"] = "." + os.pathsep + env["PYTHONPATH"]
+    except KeyError:
+        env["PYTHONPATH"] = "."
     result = subprocess.run(
         [shutil.which("python"), source_file],  # noqa: S603
         capture_output=True,
+        env=env,
     )
     assert result.returncode == 0
     assert result.stdout == b""

